File: src/main/java/by/architecture_map/belarus/entity/Source.java
package by.architecture_map.belarus.entity;

import jakarta.persistence.Entity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class Source extends BaseEntity {

    private String name;
    private String url;
    private String description;
}

File: src/main/kotlin/by/architecturemap/belarus/entity/Source.kt
package by.architecturemap.belarus.entity

import jakarta.persistence.Entity
import jakarta.validation.constraints.NotBlank

@Entity
data class Source(

    @NotBlank(message = "Name may not be blank")
    var name: String,

    @NotBlank(message = "Url may not be blank")
    var url: String,

    var description: String? = null

) : BaseEntity()
